framework/helpers/BaseArrayHelper.php

```diff
@@ -110,17 +110,14 @@ public static function toArray($object, $properties = [], $recursive = true)
      * You can use [[UnsetArrayValue]] object to unset value from previous array or
      * [[ReplaceArrayValue]] to force replace former value instead of recursive merging.
      * @param array $a array to be merged to
-     * @param array $b array to be merged from. You can specify additional
+     * @param array ...$b array to be merged from. You can specify additional
      * arrays via third argument, fourth argument etc.
      * @return array the merged array (the original arrays are not changed.)
      */
-    public static function merge($a, $b)
+    public static function merge(array $a, array $b = [])
     {
-        $args = func_get_args();
-        $res = array_shift($args);
-        while (!empty($args)) {
-            $next = array_shift($args);
-            foreach ($next as $k => $v) {
+        $merge = function (array $res, array $b) use (&$merge) {
```

Comment:
Good. I will think about this. I will be happy your advice (link to doc or tools).